File: ch4/yojkim/ch4/dollar.go
package ch4

type Dollar struct {
	amount int
}

func NewDollar(amount int) Dollar {
	dollar := Dollar{}
	dollar.amount = amount

	return dollar
}

func (d *Dollar) Times(multiplier int) *Dollar {
	return &Dollar{d.amount * multiplier}
}

func (d Dollar) Equal(input Dollar) bool {
	return d.amount == input.amount
}

File: ch4/gunddol/src/ch4/dollar.java
package ch4;

public class dollar {
    private int amount;

    public dollar(int amount){
        this.amount = amount;
    }

    public dollar times(int multiplier) {
        return new dollar(amount * multiplier);
    }

    public boolean equals(Object object){
        dollar dollar = (dollar) object;
        return amount == dollar.amount;
    }
}
